org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/LinkResourceUtil.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.managers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import org.eclipse.core.internal.preferences.EclipsePreferences;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IJavaProject;
+
+class LinkResourceUtil {
+
+	private static final IPath METADATA_FOLDER_PATH = ResourcesPlugin.getWorkspace().getRoot().getLocation().append(".projects");
+
+	private static boolean isNewer(Path file, Instant instant) throws CoreException {
+		try {
+			Instant creationInstant = Files.getFileAttributeView(file, BasicFileAttributeView.class).readAttributes().creationTime().toInstant();
+			// match accuracy
+			for (ChronoUnit unit : List.of(ChronoUnit.NANOS, ChronoUnit.MILLIS, ChronoUnit.SECONDS)) {
+				if (!creationInstant.isSupported(unit) || !instant.isSupported(unit)) {
+					creationInstant = creationInstant.truncatedTo(ChronoUnit.SECONDS);
+					instant = instant.truncatedTo(ChronoUnit.SECONDS);
```

Comment:
if ChronoUnit.NANOS is not supported, truncating to SECONDS, will render the next check against MILLIS ineffective. You need to truncated to the next precision instead

Reply:
Thanks Fred, it was indeed the intention; but poorly implemented. New patchset should cover it better.